include/AST/Decl/FunctionDecl.hpp

```diff
@@ -63,9 +84,35 @@ class FunctionDecl : public DeclBase {
     glu::types::FunctionTy *getType() const { return _type; }
 
     /// @brief Getter for the parameters of the function.
-    /// @return Returns a vector of Param objects representing the parameters of
-    /// the function.
-    llvm::ArrayRef<Param> getParams() const { return _params; }
```

Comment:
keep getParams. just return the ArrayRef from getTrailingObjects.much better than getParam and getParamsCount